hls4ml/model/hls_model.py

```diff
@@ -346,6 +346,8 @@ def insert_node(self, node, before=None):
 
         if next_node is not None:
             next_node.inputs[0] = node.outputs[0]
+        else:
+            self.outputs[0] = node.outputs[0]
```

Comment:
Models may have multiple outputs, so I think it would be safer to check if the previous node's output was also the model output and then change that instead of overwriting model's first output.

Reply:
This is now handled in #367